rte/kernels-openacc/mo_rte_solver_kernels.F90

```diff
@@ -143,12 +129,37 @@ subroutine lw_solver_noscat(ncol, nlay, ngpt, top_at_1, D, weight,
       end do
     end do
 
+    ! NOTE: This kernel produces small differences between GPU and CPU
+    ! implementations on Ascent with PGI, we assume due to floating point
+    ! differences in the exp() function. These differences are small in the
+    ! RFMIP test case (10^-6).
+    !$acc parallel loop collapse(3)
+    do igpt = 1, ngpt
+      do ilay = 1, nlay
+        do icol = 1, ncol
+          !
+          ! Optical path and transmission, used in source function and transport calculations
+          !
+          tau_loc(icol,ilay,igpt) = tau(icol,ilay,igpt)*D(icol,igpt)
+          trans  (icol,ilay,igpt) = exp(-tau_loc(icol,ilay,igpt))
+          !
+          ! Weighting factor. Use 2nd order series expansion when rounding error (~tau^2)
+          !   is of order epsilon (smallest difference from 1. in working precision)
+          !   Thanks to Peter Blossey
+          !
+          fact = merge((1._wp - trans(icol,ilay,igpt))/tau_loc(icol,ilay,igpt) - trans(icol,ilay,igpt), &
```

Comment:
I had these two kernels separated for conceptual reasons. Is the performance gain quite large?

Reply:
I well understand your conceptual reasons. From the performance point of view, that's about 1ms less out of total ~30ms.
What I could suggest is to factor out the inner code into a routine, such that this routine has no loops, just the per-point stencil operations. Then we can put the routines together in the loop and the compiler will inline them.

Reply:
Thanks to the commit I see what you mean. Is the point here to save a kernel launch? If it's really 1 part in 30 I might opt for clarity. 

The point of separate loops was that variables `trans` and `tau_loc` are used in other parts of the code as well.

Reply:
It's not just the kernel launch, it's memory reuse (`tau_loc` and `trans`). It is 1 part in 30 in the overall GPU runtime of the long-wave example, not just for this routine. The final decision is yours anyway!

Reply:
Ideally you'd like you fuse as many kernels as possible in one big loop calling point-wise routines. That can save a lot of memory traffic and decrease the runtime quite a bit, but it would require a too much of code restructuring

Reply:
I think I can live with this... maybe you can just replace the routine, rather than making a `_stencil` version?

Reply:
Sure, I went with `_stencil` simply because the other routine had a C interface and I thought you want it to be visible from outside. So I just stick to the original name, w/o `_stencil`, right?

Reply:
Ah, right. 

I have been thinking about revising the C-visible API so that only the top-level routines are visible (e.g. `lw_solver_noscat()`), but that will take a bit more time. 

So just leave things as they are.